plugins/wporg-5ftf/includes/pledge-form.php

```diff
@@ -108,6 +108,35 @@ function process_form_new() {
 		Contributor\create_new_contributor( $wporg_username, $new_pledge_id );
 	}
 
+	// Process image.
+	if ( ! function_exists('media_handle_upload') ) {
+		require_once( ABSPATH . 'wp-admin/includes/image.php' );
+		require_once( ABSPATH . 'wp-admin/includes/file.php' );
+		require_once( ABSPATH . 'wp-admin/includes/media.php' );
+	}
+
+	$logo = isset( $_FILES['org-logo'] ) ? $_FILES['org-logo'] : false;
+	if ( $logo ) {
+		if ( ! in_array( $logo['type'], [ 'image/png', 'image/jpg' ] ) ) {
+			return new WP_Error(
+				'invalid_image_type',
+				__( 'Logo file must be a png or jpg.', 'wporg' )
+			);
+		}
```

Comment:
The way that WP checks the mime type during upload is really convoluted, but `media_handle_sideload` does at some point make a call to `wp_check_filetype_and_ext`, which does some checking to try and see if the file is really what it says it is. So one alternative to this check here might be to temporarily add a filter to `upload_mimes` and set it to only allow jpg and png. Then there would be a bit more certainty about what the file actually is when it checks the type.